rust-sgx CI is currently broken

CI fails with:
```
...
          (.text+0x122): undefined reference to `__sprintf_chk'
          /usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/libcrypto.a(eng_openssl.o): in function `openssl_load_privkey':
          (.text+0x2a): undefined reference to `__fprintf_chk'
          /usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/libcrypto.a(p5_crpt.o): in function `PKCS5_PBE_keyivgen':
          (.text+0x335): undefined reference to `__memcpy_chk'
          /usr/bin/ld: (.text+0x354): undefined reference to `__memcpy_chk'
          /usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/libcrypto.a(v3_alt.o): in function `i2v_GENERAL_NAME':
          (.text+0x2f1): undefined reference to `__strcat_chk'
          /usr/bin/ld: (.text+0x308): undefined reference to `__strcat_chk'
          /usr/bin/ld: (.text+0x339): undefined reference to `__strcat_chk'
          /usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/libcrypto.a(p5_scrypt.o): in function `PKCS5_pbe2_set_scrypt':
          (.text+0x138): undefined reference to `__memcpy_chk'
          collect2: error: ld returned 1 exit status

  = note: some `extern` functions couldn't be found; some native libraries may need to be installed or have their path specified
  = note: use the `-l` flag to specify native libraries to link
  = note: use the `cargo:rustc-link-lib` directive to specify the native libraries to link with Cargo (see https://doc.rust-lang.org/cargo/reference/build-scripts.html#cargorustc-link-libkindname)

error: could not compile `fortanix-sgx-tools` (bin "ftxsgx-runner") due to previous error

The command "mkdir -p /tmp/muslinclude && ln -sf /usr/include/x86_64-linux-gnu/openssl /tmp/muslinclude/openssl && PKG_CONFIG_ALLOW_CROSS=1 CFLAGS=-I/tmp/muslinclude cargo build --locked -p fortanix-sgx-tools --target x86_64-unknown-linux-musl" exited with 101.
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -58,9 +58,8 @@ matrix:
         #  fault while compiling. We only execute `faketime` during the tests
         #- cargo test --locked -p nitro-attestation-verify --no-run && faketime '2021-09-08 11:00:00 GMT' cargo test --locked -p nitro-attestation-verify --lib
         - cargo test --locked -p nitro-attestation-verify --no-run && faketime '2021-09-10 11:00:00 GMT' cargo test --locked -p nitro-attestation-verify --lib
-        # NOTE: linking glibc version of OpenSSL with musl binary.
-        # Unlikely to produce a working binary, but at least the build succeeds.
-        - mkdir -p /tmp/muslinclude && ln -sf /usr/include/x86_64-linux-gnu/openssl /tmp/muslinclude/openssl && PKG_CONFIG_ALLOW_CROSS=1 CFLAGS=-I/tmp/muslinclude cargo build --locked -p fortanix-sgx-tools --target x86_64-unknown-linux-musl
+        # NOTE: Skipping linking with the glibc version of OpenSSL to produce a musl based binary. It is unlikely that this would produce a working binary anyway.
+        - mkdir -p /tmp/muslinclude && ln -sf /usr/include/x86_64-linux-gnu/openssl /tmp/muslinclude/openssl && PKG_CONFIG_ALLOW_CROSS=1 CFLAGS=-I/tmp/muslinclude CARGO_TARGET_X86_64_UNKNOWN_LINUX_MUSL_LINKER=true cargo build --locked -p fortanix-sgx-tools --target x86_64-unknown-linux-musl
         - cargo build --verbose --locked -p em-app -p get-certificate -p harmonize --target=x86_64-unknown-linux-musl
         - cargo build --verbose --locked -p em-app -p get-certificate -p harmonize --target=x86_64-fortanix-unknown-sgx
         - ./doc/generate-api-docs.sh

